File: MyPersonalIndex/Classes/Queries/AcctQueries.cs
using System.Data;
using System.Data.SqlServerCe;

namespace MyPersonalIndex
{
    class AcctQueries: Queries
    {
        public static QueryInfo DeleteAcct(int ID)
        {
            return new QueryInfo(
                "DELETE FROM Accounts WHERE ID = @ID",
                new SqlCeParameter[] { 
                    AddParam("@ID", SqlDbType.Int, ID)
                }
            );
        }

        public static QueryInfo InsertAcct(int Portfolio, string Name, double? TaxRate, bool OnlyGain)
        {
            return new QueryInfo(
                "INSERT INTO Accounts (Portfolio, Name, TaxRate, OnlyGain) VALUES (@Portfolio, @Name, @TaxRate, @OnlyGain)",
                new SqlCeParameter[] {
                    AddParam("@Portfolio", SqlDbType.Int, Portfolio),
                    AddParam("@Name", SqlDbType.NVarChar, Name),
                    AddParam("@TaxRate", SqlDbType.Decimal, TaxRate.HasValue ? TaxRate : (object)System.DBNull.Value),
                    AddParam("@OnlyGain", SqlDbType.Bit, OnlyGain)
                }
            );
        }

        public static QueryInfo UpdateAcct(int ID, string Name, double? TaxRate, bool OnlyGain)
        {
            return new QueryInfo(
                   "UPDATE Accounts SET Name = @Name, TaxRate = @TaxRate, OnlyGain = @OnlyGain WHERE ID = @ID",
                   new SqlCeParameter[] {
                    AddParam("@ID", SqlDbType.Int, ID),
                    AddParam("@Name", SqlDbType.NVarChar, Name),
                    AddParam("@TaxRate", SqlDbType.Decimal, TaxRate.HasValue ? TaxRate : (object)System.DBNull.Value),
                    AddParam("@OnlyGain", SqlDbType.Bit, OnlyGain)
                }
            );
        }
    }
}

File: MyPersonalIndex/classes/queries/acctQueries.cpp
#include "acctQueries.h"
#include "queries.h"
#include "globals.h"

queries::queryInfo* acctQueries::updateAcct(globals::account* const acct)
{
    if (!acct)
        return 0;

    QVariant taxRate(QVariant::Double);
    if (acct->taxRate >= 0)
        taxRate = acct->taxRate;

    if(acct->id == -1) // insert new
    {
        return new queryInfo(
            "INSERT INTO Acct (PortfolioID, Description, TaxRate, TaxDeferred)"
            " VALUES (:PortfolioID, :Description, :TaxRate, :TaxDeferred)",
            QList<parameter>()
                << parameter(":PortfolioID", m_portfolioID)
                << parameter(":Description", acct->description)
                << parameter(":TaxRate", taxRate)
                << parameter(":TaxDeferred", (int)acct->taxDeferred)
        );
    }
    else // update
    {
        return new queryInfo(
            "UPDATE Acct SET Description = :Description, TaxRate = :TaxRate, TaxDeferred = :TaxDeferred WHERE ID = :AcctID",
            QList<parameter>()
                << parameter(":Description", acct->description)
                << parameter(":TaxRate", taxRate)
                << parameter(":TaxRate", taxRate)
                << parameter(":AcctID", acct->id)
        );
    }
}

queries::queryInfo* acctQueries::deleteAcct(const int &acctID)
{
    return new queryInfo(
            "DELETE FROM Acct WHERE ID = :AcctID",
            QList<parameter>()
                << parameter(":AcctID", acctID)
        );
}
